### ISSUE TITLE
PycurlClient does not support TCP keep-alive

### ISSUE THREAD

**Issue by href** [stripe/stripe-python#324]
The RequestsClient uses a session with which one is able to get TCP keep-alive by creating a single client per process:

`stripe.default_http_client = stripe.http_client.RequestsClient()`

After the initial stripe API handshake, the API calls can be completed much quicker. For long running processes (say a web-application) this yields a great speed up. In our case it cuts the costs of API calls in half.

Doing the same thing with the PycurlClient doesn't work however. The following setup will still discard the connection after every single request:

`stripe.default_http_client = stripe.http_client.PycurlClient()`

To facilitate this the Curl instance would have to be kept around if I'm not mistaken. So basically the PycurlClient might want to use curl as a kind of session, just like the RequestsClient does:

```python
class PycurlClient(object):
    def __init__(..., curl=None):
        self.curl = curl or pycurl.Curl()
    def request(...):
        curl = self.curl
```

Python Version: 3.5.3
Library Version: 1.57.1

**Comment by brandur-stripe** [comment]
Yes, this seems like a good idea. Added a fix in #325.

**Comment by brandur-stripe** [comment]
Released as 1.58.0.
